packages/hardhat-core/src/internal/cli/cli.ts

```diff
@@ -22,14 +23,20 @@ import { isRunningOnCiServer } from "../util/ci-detection";
 import {
   hasConsentedTelemetry,
   writeTelemetryConsent,
+  hasPromptedForExtension,
+  writePromptedForExtension,
 } from "../util/global-dir";
 import { getPackageJson, PackageJson } from "../util/packageInfo";
 
 import { applyWorkaround } from "../util/antlr-prototype-pollution-workaround";
 import { Analytics } from "./analytics";
 import { ArgumentsParser } from "./ArgumentsParser";
 import { enableEmoji } from "./emoji";
-import { confirmTelemetryConsent, createProject } from "./project-creation";
+import { createProject } from "./project-creation";
+import {
+  confirmTelemetryConsent,
+  confirmExtensionInstallation,
```

Comment:
Similarly, I'd like to see this identifier renamed to answer the question "_what_ extension?" Perhaps `confirmHHVSCodeInstallation`?